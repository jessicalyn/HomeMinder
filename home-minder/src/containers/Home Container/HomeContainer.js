import React, { Component } from 'react'
import { connect } from 'react-redux'
import HomeCard from '../HomeCard/HomeCard'

export class HomeContainer extends Component {

  // displayAllItems = () => {
  //   return this.props.items.map(item => {
  //     return <HomeCard key={ item.id } item={ item } />
  //   })
  // }

  render() {
    return(
      <div>
        
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  items: state.items
})

export default connect(mapStateToProps)(HomeContainer)