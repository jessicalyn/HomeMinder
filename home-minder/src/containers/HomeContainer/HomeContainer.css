.home-container {
  background-color: white;
  border: solid 2px black;
  border-radius: 2rem;
  width: 75%;
  margin: auto;
  margin-top: 2rem;
  padding: 1.5rem
}