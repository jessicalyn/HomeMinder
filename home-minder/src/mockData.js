export const mockHomeData = [
  { id: "JK1234",
    room: "Kitchen",
    category: "Filters",
    name: "Refrigerator Water Filter",
    brand: "",
    model: "",
    lastReplaced: "",
    purchaseLocation: "",
    recommendation: "6 months",
    userScheduled: "",
    type: "Replace"
  },
  { id: "IF1234",
    room: "Kitchen",
    category: "Sponges",
    name: "Sponge",
    brand: "",
    model: "",
    lastReplaced: "",
    purchaseLocation: "",
    recommendation: "2-4 weeks",
    userScheduled: "",
    type: "Replace"
  },
  { id: "PF123",
    room: "Bathroom",
    category: "Toothbrush",
    name: "Toothbrush",
    brand: "",
    model: "",
    lastReplaced: "",
    purchaseLocation: "",
    recommendation: "1-3 months",
    userScheduled: "",
    type: "Replace"
  },
]