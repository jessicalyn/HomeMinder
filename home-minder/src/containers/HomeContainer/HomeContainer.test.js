import React from 'react';
import { HomeContainer, mapStateToProps } from './HomeContainer';
import { shallow } from 'enzyme';
import { mockHomeData } from '../../mockData'

describe('HomeContainer container', () => {
  describe('HomeContainer', () => {
    let wrapper
    const mockItems = mockHomeData

    beforeEach(() => {
      wrapper = shallow(<HomeContainer items={ mockItems }/>)
    });

    it('should match the snapshot with the data passed in', () => {
      expect(wrapper).toMatchSnapshot()
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with an items array', () => {
      const mockState = {
        items: mockHomeData,
        error: "",
        loading: false
      }
      const expected = {
        items: mockHomeData
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  });

})