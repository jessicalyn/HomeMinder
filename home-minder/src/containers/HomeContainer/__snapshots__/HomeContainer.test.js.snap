// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer container HomeContainer should match the snapshot with the data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeContainer
    items={
      Array [
        Object {
          "brand": "",
          "category": "Filters",
          "id": "JK1234",
          "lastReplaced": "",
          "model": "",
          "name": "Refrigerator Water Filter",
          "purchaseLocation": "",
          "recommendation": "6 months",
          "room": "Kitchen",
          "type": "Replace",
          "userScheduled": "",
        },
        Object {
          "brand": "",
          "category": "Sponges",
          "id": "IF1234",
          "lastReplaced": "",
          "model": "",
          "name": "Sponge",
          "purchaseLocation": "",
          "recommendation": "2-4 weeks",
          "room": "Kitchen",
          "type": "Replace",
          "userScheduled": "",
        },
        Object {
          "brand": "",
          "category": "Toothbrush",
          "id": "PF123",
          "lastReplaced": "",
          "model": "",
          "name": "Toothbrush",
          "purchaseLocation": "",
          "recommendation": "1-3 months",
          "room": "Bathroom",
          "type": "Replace",
          "userScheduled": "",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Schedule reminders for items around your home
        </h3>,
        <div>
          <h4>
            Upcoming Reminders
          </h4>
        </div>,
      ],
      "className": "home-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Schedule reminders for items around your home",
        },
        "ref": null,
        "rendered": "Schedule reminders for items around your home",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Upcoming Reminders
            </h4>,
            Array [],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Upcoming Reminders",
            },
            "ref": null,
            "rendered": "Upcoming Reminders",
            "type": "h4",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Schedule reminders for items around your home
          </h3>,
          <div>
            <h4>
              Upcoming Reminders
            </h4>
          </div>,
        ],
        "className": "home-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Schedule reminders for items around your home",
          },
          "ref": null,
          "rendered": "Schedule reminders for items around your home",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Upcoming Reminders
              </h4>,
              Array [],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Upcoming Reminders",
              },
              "ref": null,
              "rendered": "Upcoming Reminders",
              "type": "h4",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
